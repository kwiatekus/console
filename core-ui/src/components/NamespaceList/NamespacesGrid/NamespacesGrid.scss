.grid-wrapper {
    display: grid;
    grid-template-columns: repeat( auto-fit, 260px );
    grid-gap: 16px;
}
