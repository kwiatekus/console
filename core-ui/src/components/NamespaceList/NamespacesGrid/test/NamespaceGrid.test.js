import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing';

import NamespacesGrid from './../NamespacesGrid';

const sampleNamespaces = [
  {
    name: 'a',
    status: 'Active',
    pods: [{ status: 'ACTIVE' }],
    isSystemNamespace: false,
    applications: [{}, {}],
  },
  {
    name: 'b',
    status: 'Active',
    pods: [{ status: 'ACTIVE' }],
    isSystemNamespace: true,
    applications: [{}, {}, {}],
  },
];

describe('NamespaceGrid', () => {
  it('Renders with minimal props', () => {
    const component = renderer.create(
      <MockedProvider>
        <NamespacesGrid namespaces={sampleNamespaces} />
      </MockedProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
