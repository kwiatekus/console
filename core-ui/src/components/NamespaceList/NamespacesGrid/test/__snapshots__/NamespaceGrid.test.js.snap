// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamespaceGrid Renders with minimal props 1`] = `
<ul
  className="grid-wrapper fd-has-margin-medium"
>
  <li>
    <div
      className="fd-panel namespace-details-card"
      onClick={[Function]}
    >
      <div
        className="fd-panel__header fd-has-color-text-1 fd-has-type-1"
      >
        <span>
          a
        </span>
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <button
                className="fd-button--light sap-icon--overflow popover-control"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="fd-panel__body"
      >
        <section
          className="namespace-details-card__body"
        >
          <div>
            <p
              className="fd-has-type-4 fd-has-font-weight-light"
              style={
                Object {
                  "color": "#bb0000",
                }
              }
            >
              0
              /
              1
            </p>
            <p>
              Pods are healthy
            </p>
          </div>
          <div>
            <p
              className="fd-has-type-4 fd-has-font-weight-light"
            >
              2
            </p>
            <p>
              Bound Applications
            </p>
          </div>
        </section>
        <p
          className="fd-has-type-minus-1 fd-has-color-text-2 fd-has-margin-top-tiny"
        />
      </div>
    </div>
  </li>
  <li>
    <div
      className="fd-panel namespace-details-card"
      onClick={[Function]}
    >
      <div
        className="fd-panel__header fd-has-color-text-1 fd-has-type-1"
      >
        <span>
          b
          <span
            className="fd-badge fd-has-margin-left-tiny"
          >
            System
          </span>
        </span>
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <button
                className="fd-button--light sap-icon--overflow popover-control"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="fd-panel__body"
      >
        <section
          className="namespace-details-card__body"
        >
          <div>
            <p
              className="fd-has-type-4 fd-has-font-weight-light"
              style={
                Object {
                  "color": "#bb0000",
                }
              }
            >
              0
              /
              1
            </p>
            <p>
              Pods are healthy
            </p>
          </div>
          <div>
            <p
              className="fd-has-type-4 fd-has-font-weight-light"
            >
              3
            </p>
            <p>
              Bound Applications
            </p>
          </div>
        </section>
        <p
          className="fd-has-type-minus-1 fd-has-color-text-2 fd-has-margin-top-tiny"
        />
      </div>
    </div>
  </li>
</ul>
`;
